iocore/net/SSLSNIConfig.cc

```diff
@@ -111,40 +112,63 @@ int
 SNIConfigParams::load_sni_config()
 {
   for (auto &item : yaml_sni.items) {
-    auto ai = sni_action_list.emplace(sni_action_list.end());
-    ai->set_glob_name(item.fqdn);
-    Debug("ssl", "name: %s", item.fqdn.data());
+    Debug("ssl_sni", "name: %s", item.fqdn.data());
+
+    ActionVector *actions = nullptr;
+
+    if (item.fqdn.find('*') == std::string::npos) {
+      // servername is case-insensitive, store & find it in lower case
+      char lower_case_name[TS_MAX_HOST_NAME_LEN + 1];
+      ts::transform_lower(item.fqdn, lower_case_name);
+      auto [it, inserted] = sni_action_map.emplace(std::make_pair(lower_case_name, ActionVector()));
+
+      if (!inserted) {
+        Error("error on loading sni yaml - fqdn=%s", item.fqdn.data());
```

Comment:
It's not a good habit to use data() as a substitute for c_str().  If you don't mean to change the string contents, c_str() makes that clear (to maintainers and the compiler).

Reply:
Good point 👍 I thought it's `std::string_view`, but it's `std::sting`.